srcStatic/ConsoleModuleLoader.h

```diff
@@ -5,9 +5,7 @@
 
 class ConsoleModuleLoader {
 public:
-	ConsoleModuleLoader();
-	// Test Constructor
-	ConsoleModuleLoader(const std::string& testModuleDirectory);
+	ConsoleModuleLoader(const std::string& moduleRelativeDirectory);
```

Comment:
Correct assumption. Although for non-unit-test code, you can assume it is the game's executable. I'm not opposed to allowing absolute paths. It is another feature to design and test though. I guess we can open an issue if you would like the feature. Forcing to pass a relative has worked so far.

Reply:
Actually, after some investigation, I think absolute paths might already be supported. The relevant line for this is:
```cpp
moduleDirectory = fs::path(GetGameDirectory()).append(moduleRelativeDirectory).string();
```
Checking the documentation for [`path::append`](https://en.cppreference.com/w/cpp/filesystem/path/append), it seems when the path being appended is absolute, it replaces the base path, rather than being added to it.

Hence, relative paths will be resolved relative to `GetGameDirectory()`. Absolute paths should be used as is.

This may be a case of just writing a few unit tests with absolute paths to verify current behavior, and perhaps documenting it.

Reply:
This is on hold unless we can get the variables `std::string moduleDirectory;` and `std::string moduleName;` as members of ConsoleModuleDirectory. Currently they are not part of the class and they persist between test instances of ConsoleModuleDirectory. I tried adding a unit test for this, but it is causing problems due to the persistence. It is a bug in the rest of the code, but didn't show up until expanding the unit tests to cover this. 

I made a forum post to ask for help in resolving as there are some specific technical requirements to fix.

Reply:
I put a hack in that resets the moduleDirectory every time ConsoleModuleLoader is re-initialized. 

Perhaps ConsoleModuleLoader should be a static class.

To support absolute paths, someone has to write a function that retrieves the final directory in the path. Otherwise the module name would be the absolute path.